lib/cylc/mp_pool.py

```diff
@@ -159,7 +163,14 @@ def process(self):
         for proc, ctx, callback, callback_args in self.runnings:
             ret_code = proc.poll()
             if ret_code is None:
-                runnings.append([proc, ctx, callback, callback_args])
+                if time.time() > ctx.init_time + self.cmd_timeout:
```

Comment:
Worth moving time.time() to before line 163 to reduce the number of system calls.

```python
now = time.time()
for proc, ctx, callback, callback_args in self.runnings:
    ret_code = proc.poll()
    if ret_code is None:
        if now > ctx.init_time + self.cmd_timeout: